engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/SafeHtmlValidatorTest.java

```diff
@@ -164,6 +164,20 @@ public void testAdditionalProtocols() {
 		assertNumberOfViolations( validator.validate( new Bar( "<div src='data:image/png;base64,100101' />" ) ), 1 );
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HV-817")
+	public void testPreserveRelativeLinks() throws Exception {
+		descriptor.setValue( "whitelistType", WhiteListType.RELAXED );
+		descriptor.setValue( "baseUri", "http://127.0.0.1" );
+
+		assertTrue( getSafeHtmlValidator().isValid( "<img src='/some/relative/url/image.png' />", null ) );
```

Comment:
@gsmet It seems jsoup does not fail validation if the relative path goes back too far. jsoup's `StringUtil.resolve(URL base, String relUrl)` uses `new java.net.URL(base, relUrl)`, which will resolve `http://localhost/` with `../../../../../test.png` as URL containing `http://localhost/../test.png`. It is not examined further. No active issues with jsoup in this area, so we might consider raising one - what do you think?

If the baseURI default value is set by default, the default behaviour of SafeHtml would change, since it currently does not allow relative paths.